src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala

```diff
@@ -374,10 +418,7 @@ final case class ClientApiServiceImpl(
       descriptor <- eService.descriptors
         .find(_.id == agreement.descriptorId)
         .toFuture(DescriptorNotFound(purpose.eserviceId, agreement.descriptorId))
-      version    <- purpose.versions
-        .filterNot(v => invalidPurposeStates(v.state))
```

Comment:
I may be missing something, but isn't enough to add to just add the statuses `Draft, Archived, Rejected, WaitingForApproval` to `invalidPurposeStates`, instead of adding the additional code?

Reply:
If @beetlecrunch agrees then I can add the states and remove code

Reply:
Ok, I agree with @galales .
Let's simply extend `invalidPurposeStates` .